packages/libs/error-stack/src/fmt/mod.rs

```diff
@@ -0,0 +1,1075 @@
+//! // value, we use `HookContext` here as storage, to store values specific to this hook.
+//! // Here we make use of the auto-incrementing feature.
+//! // The incrementation is type specific, meaning that `ctx.increment()` for the `u32` hook will
+//! // not influence the counter of the `u64` or `u16` hook.
+//! Report::install_debug_hook::<u32>(|_, ctx| Emit::next(format!("u32 value {}", ctx.increment())));
+//!
+//! // we do not need to make use of the context, to either store a value for the duration of the
+//! // rendering, or to render additional text, which is why we omit the parameter.
+//! Report::install_debug_hook::<u64>(|val, _| Emit::next(format!("u64 value ({val})")));
+//!
+//! Report::install_debug_hook::<u16>(|_, ctx| {
+//!     // we set a value, which will be removed on non-alternate views
+//!     // and is going to be appended to the actual return value.
+//!     if ctx.alternate() {
+//!         ctx.attach_snippet("Look! I was rendered from a `u16`");
+//!     }
+//!     Emit::next("For more information, look down below")
+//!  });
+//!
+//! // you can use arbitrary values as arguments, just make sure that you won't repeat them.
+//! // here we use [`Emit::defer`], this means that this value will be put at the end of the group.
+//! Report::install_debug_hook::<String>(|val, _| Emit::defer(val));
+//!
+//! let report = Report::new(Error::from(ErrorKind::InvalidInput))
+//!     .attach(2u64)
+//!     .attach("This is going to be at the end".to_owned())
+//!     .attach(3u32)
+//!     .attach(3u32)
+//!     .attach(4u16);
+//!
+//! # owo_colors::set_override(true);
+//! # fn render(value: String) -> String {
+//! #     let backtrace = regex::Regex::new(r"Backtrace No\. (\d+)\n(?:  .*\n)*  .*").unwrap();
+//! #     let backtrace_info = regex::Regex::new(r"backtrace with (\d+) frames \((\d+)\)").unwrap();
+//! #
+//! #     let value = backtrace.replace_all(&value, "Backtrace No. $1\n  [redacted]");
+//! #     let value = backtrace_info.replace_all(value.as_ref(), "backtrace with [n] frames ($2)");
+//! #
+//! #     ansi_to_html::convert_escaped(value.as_ref()).unwrap()
+//! # }
+//! #
+//! # #[cfg(nightly)]
+//! # expect_test::expect_file![concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt__doc.snap")].assert_eq(&render(format!("{report:?}")));
+//! #
+//! println!("{report:?}");
+//!
+//! # #[cfg(nightly)]
+//! # expect_test::expect_file![concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt_doc_alt.snap")].assert_eq(&render(format!("{report:#?}")));
+//! #
+//! println!("{report:#?}");
+//! ```
+//! ### `println!("{report:?}")`
+//!
+//! <pre>
+#![doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt__doc.snap"))]
+//! </pre>
+//!
+//! ### `println!("{report:#?}")`
+//!
+//! <pre>
+#![doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/doc/fmt_doc_alt.snap"))]
+//! </pre>
```

Comment:
The output of the alternate representation is the same as the non-alternate one.

Reply:
fixed in 93a150aa4c1193cae65f69f3e877034c1faf63e9